Create auto-generated objects.go file

achernevskii opened the pull request:
# Issue/Feature description

* Structs for WAPI objects should be auto-generated to simplify the go-client usage and development.

# How it was fixed/implemented

* ObjectType field of the IBObject is replaced with the ObjectType
  method for each child object.

* Override ReturnFields method for each struct. By default, it will return
  a set of standard fields for each respective struct.

* Removed Ipv4Addr/Ipv6Addr fields from the HostRecord object.

* Updated existing object manager unit tests.

* The IBObjectManager interface is not changed.

# Tests

* Created a E2E connector facade, that will remove all objects created by e2e test after the test is done.

* Created a set of E2E tests to check crud operations on the generated object structs.

# Reviewers

@skudriavtsev

achernevskii commented on utils/utils.go:
```diff
@@ -0,0 +1,6 @@
+package utils
+
+// Bool returns a pointer to the bool value passed as argument
+func Bool(v bool) *bool {
```
`PtrBool`

skudriavtsev reviewed (changes requested):
The major comments:

1. 'omitempty' attribute will be harmfull in some cases; for example, if 'comment' field was set and a user wants to make it empty, it will not be possible to do using the  implementation from this PR. As we discussed, this may be solved using a pointer to field's value instead of the value itself in the structure.
2. It is hard to maintain the code located in a single file, at least when two or more people work on different issues//features but the code is in the same file. We can avoid this splitting the implementation into several files, one for each object type, for example.

Minor comment. It would be better to put an empty line between a field's definition in a structure and a comment for the next field in this structure. Just to improve readability.

achernevskii commented:
Hello @skudriavtsev,

> 'omitempty' attribute will be harmfull in some cases;

I couldn't think of a scenario, in which `omitempty` attribute could be "harmful". However it restricts a library user from using default values (e.g. "" for string) as some field values. That's inconvenient, so I've changed all the updatable field's types to pointers. This way nil pointer will mean an absence of a value, and pointer to an empty string in case of a string will mean an empty string value.

> It is hard to maintain the code located in a single file

This code is meant to be auto-generated. If somebody needs to change this file, this person should update the generator configuration first, and then use the generator to update the file. In this case, I don't think that it's gonna be hard to maintain the code, since there are no merge conflicts, when we have the generator which should always create the same code, if configuration isn't changed.  Also it's much more easier to explain developers that they can't manually update one file, instead of articulating the same thought about the 100+ auto-generated files.

> It would be better to put an empty line between a field's definition in a structure and a comment for the next field in this structure. Just to improve readability.

Added empty lines between field definitions.

skudriavtsev commented:
At least, 'comment' field is not fixed: updating with an empty comment leads to no update at all. At many objects.

Besides, 'extattrs' field is also marked as 'omitempty', and this leads to the situation when we try to update with an empty list of EAs (removing all EAs) but update does not happen. This behavior must be changed.

achernevskii commented:
> At least, 'comment' field is not fixed: updating with an empty comment leads to no update at all. At many objects.

@skudriavtsev, could you provide the object with a problem `comment` field?

Tested it out on the `View` object, `comment` field is updatable when I specify the empty string. You can see the test case here: 819584b38b0f8b31bc1239dd46c60017f9ebc1f7.

However you should know, that API will not return an empty string, when comment is set to empty string, even if you specify comment in `_return_fields`. This is a WAPI behaviour, that cannot and shouldn't be changed on the client's side:
![image](https://user-images.githubusercontent.com/91904219/214452016-e76efd12-1d9d-476b-bac0-1d0d436f5d5b.png)

skudriavtsev commented:
Sorry, it seems that I have not retrieved the latest version of the code from your repository :-( I will re-test.

achernevskii commented:
> Besides, 'extattrs' field is also marked as 'omitempty', and this leads to the situation when we try to update with an empty list of EAs (removing all EAs) but update does not happen. This behavior must be changed.

Changed EA behavior in 0e3d85267819f79004ae047c9f725ebffd5cbad9. Now one can remove all extensible attributes via update operation.
